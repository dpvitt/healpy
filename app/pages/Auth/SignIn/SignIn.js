import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Input, Button } from 'react-native-elements';

export const SignIn = ({ setEmail, setPassword, handleSignIn }) => (
  <Fragment>
    <Input
      label="Email"
      leftIcon={{ type: 'font-awesome', name: 'envelope' }}
      onChangeText={(value) => setEmail(value)}
      placeholder="my@email.com"
    />
    <Input
      label="Password"
      leftIcon={{ type: 'font-awesome', name: 'lock' }}
      onChangeText={(value) => setPassword(value)}
      placeholder="p@ssw0rd123"
      secureTextEntry
    />
    <Button title="Submit" onPress={handleSignIn} />
  </Fragment>
);

SignIn.propTypes = {
  setEmail: PropTypes.func.isRequired,
  setPassword: PropTypes.func.isRequired,
  handleSignIn: PropTypes.func.isRequired,
};
