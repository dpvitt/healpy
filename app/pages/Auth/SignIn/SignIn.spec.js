import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from './SignIn';

describe('<SignIn />', () => {
  const props = {
    setEmail: jest.fn(),
    setPassword: jest.fn(),
    handleSignIn: jest.fn(),
  };

  const render = shallow(<SignIn {...props} />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    expect(render).toMatchSnapshot();
  });

  it('should call setEmail when email input updated', () => {
    const emailInput = render.find('Input').at(0);
    emailInput.prop('onChangeText')('test');
    expect(props.setEmail).toHaveBeenCalledWith('test');
  });

  it('should call setPassword when password input updated', () => {
    const passwordInput = render.find('Input').at(1);
    passwordInput.prop('onChangeText')('test');
    expect(props.setPassword).toHaveBeenCalledWith('test');
  });

  it('should call handleSignIn when button pressed', () => {
    const button = render.find('Button');
    button.prop('onPress')();
    expect(props.handleSignIn).toHaveBeenCalled();
  });
});
