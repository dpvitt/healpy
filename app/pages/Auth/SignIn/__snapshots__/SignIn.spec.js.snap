// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should render 1`] = `
<Fragment>
  <Input
    label="Email"
    leftIcon={
      Object {
        "name": "envelope",
        "type": "font-awesome",
      }
    }
    onChangeText={[Function]}
    placeholder="my@email.com"
  />
  <Input
    label="Password"
    leftIcon={
      Object {
        "name": "lock",
        "type": "font-awesome",
      }
    }
    onChangeText={[Function]}
    placeholder="p@ssw0rd123"
    secureTextEntry={true}
  />
  <Button
    onPress={[MockFunction]}
    title="Submit"
  />
</Fragment>
`;
